### PR TITLE
Adding the GEV and GP extreme distributions to Show model and generate random Samples Dialogues

### PR DESCRIPTION
**Pull request by Ivanluv**
This PR adds GEV and GP distributions to the show dialogue and the generate random dialogue.@africanmathsinitiative/developers this is ready for review

### CODE REVIEW

**Review by rdstern — APPROVED** [review]
These changes seem fine

**Comment by lloyddewit** [comment]
@shadrackkibet please could you review? thanks

File: instat/static/InstatObject/R/stand_alone_functions.R [hunk 1]
```diff
@@ -1282,3 +1282,40 @@ summary_sample <- function(x, size, replace = FALSE){
...
+qevd <- function (p, loc = 0, scale = 1, shape = 0, threshold = 0, Type = c("GEV", 
+                                                                            "GP", "PP", "Gumbel", "Frechet", 
+                                                                            "Weibull", "Exponential", "Beta", "Pareto"), 
+                  lower.tail = TRUE) 
+{
+  Type <- match.arg(Type)
+  Type <- tolower(Type)
+  if (scale <= 0) 
+    stop("qevd: invalid scale argument.  Must be > 0.")
+  #if (min(p, na.rm = TRUE) <= 0 || max(p, na.rm = TRUE) >= 1) stop("qevd: invalid p argument.  Must have 0 < p < 1.")
```
Comment (shadrackkibet) [inline instat/static/InstatObject/R/stand_alone_functions.R:1296]: We need to understand why this line of code is causing issues.

**Review by rdstern — CHANGES_REQUESTED** [review]
I have found a small problem.  This dialogue can be used without a data frame being present, i.e. with "no data loaded".  This gives an error, but then works and produces the graph.  Unless it is trivial to fix, I suggest we don't allow OK to be enabled in this dialogue unless there is a data frame.  

Once merged, there remain a lot of further changes to the probabilities dialogue.

**Comment by lloyddewit** [comment]
@Ivanluv are you able to implement the change @rdstern requested?

**Review by rdstern — CHANGES_REQUESTED** [review]
This also needs an update because of the version of R

**Review by rdstern — CHANGES_REQUESTED** [review]
@Ivanluv this needs the update for the version of R.

**Comment by Ivanluv** [comment]
@rdstern it already has conflicts .I am to work on the issue again and close this PR

**Comment by shadrackkibet** [comment]
@Ivanluv any progress here?

**Comment by Ivanluv** [comment]
@shadrackkibet currently none .I don't understand the R changes to be implemented for this to work. I therefore asked @rdstern to have this assigned to someone else

**Comment by rdstern** [comment]
I thought that was ok, i.e. show model and random samples.  What remains is adding these distributions to the Fit > One Variable dialogue.  That may be implicit here or a different issue.

In the one variable fit, this was waiting until the test and estimate were merged - which has now happened.

**Comment by lloyddewit** [comment]
@Ivanluv I'm not sure how to interpret the comments above. Are you able to progress this (potentially with support) or should we try and find someone else to take over?
thanks for clarifying

**Comment by rdstern** [comment]
I was assuming this would be done by @shadrackkibet

**Comment by shadrackkibet** [comment]
Happy to take over. Thanks.

**Comment by Ivanluv** [comment]
@shadrackkibet take over

File: instat/My Project/Resources.resx [hunk 2]
```diff
@@ -1645,4 +1645,13 @@
...
   </data>
+  <data name="Generalized_Extreme_Value" xml:space="preserve">
+    <value>Generalized Extreme Value</value>
+  </data>
+  <data name="Generalized_Pareto_Distribution" xml:space="preserve">
+    <value>Generalized Pareto Distribution</value>
+  </data>
+  <data name="Gumbel" xml:space="preserve">
+    <value>Gumbel</value>
+  </data>
```
Comment (shadrackkibet) [inline instat/My Project/Resources.resx:1656]: @Ivanluv  are you able to remove changes from these files? This will ensure a smooth takeover.

Comment (Ivanluv) [reply instat/My Project/Resources.resx:1656]: Let me do so

Comment (Ivanluv) [reply instat/My Project/Resources.resx]: Done

File: instat/My Project/Resources.Designer.vb [hunk 3]
```diff
@@ -22,7 +22,7 @@ Namespace My.Resources
     '''<summary>
     '''  A strongly-typed resource class, for looking up localized strings, etc.
     '''</summary>
-    <Global.System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "15.0.0.0"),  _
+    <Global.System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "16.0.0.0"),  _
```
Comment (shadrackkibet) [inline instat/My Project/Resources.Designer.vb:25]: What about this?

File: instat/My Project/Resources.Designer.vb [hunk 4]
```diff
@@ -14,18 +14,18 @@ Option Explicit On
...
     '''  A strongly-typed resource class, for looking up localized strings, etc.
     '''</summary>
-    <Global.System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "15.0.0.0"),  _
-     Global.System.Diagnostics.DebuggerNonUserCodeAttribute(),  _
-     Global.System.Runtime.CompilerServices.CompilerGeneratedAttribute(),  _
-     Global.Microsoft.VisualBasic.HideModuleNameAttribute()>  _
+    <Global.System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "15.0.0.0"),
+     Global.System.Diagnostics.DebuggerNonUserCodeAttribute(),
+     Global.System.Runtime.CompilerServices.CompilerGeneratedAttribute(),
+     Global.Microsoft.VisualBasic.HideModuleNameAttribute()>
```
Comment (shadrackkibet) [inline instat/My Project/Resources.Designer.vb:28]: I still see some changes in this file.

**Comment by lloyddewit** [comment]
This PR is replaced by PR #6214

**Pull request closed without merging**